### PR TITLE
Fixed | Nested query with bool doesn't work when there is only one MU…

### PR DESCRIPTION
**Pull request by trandangtri**
Related to this bug https://github.com/ongr-io/ElasticsearchDSL/issues/32

### CODE REVIEW

**Comment by saimaz** [comment]
@trandangtri next time when you do a PR please include a closing comment for a specific issue if there is one. e.g.

Closing #32

**Comment by saimaz** [comment]
@trandangtri please provide a few unit test cases to represent a problem and to make sure our fix works well.

We should test:
- [ ] Case with 1 query in the must condition
- [ ] Case with 1 queries in the should condition
- [ ] Case with 2 queries in the must condition
- [ ] Case with 2 queries in the should condition
- [ ] Case with 1 query in each, must, must_not and should conditions

File: tests/Query/NestedQueryTest.php [hunk 1]
```diff
@@ -11,27 +11,32 @@
...
+
     /**
      * Tests toArray method.
      */
     public function testToArray()
     {
         $missingFilterMock = $this->getMockBuilder('ONGR\ElasticsearchDSL\Filter\MissingFilter')
-            ->setConstructorArgs(['test_field'])
-            ->getMock();
+                                  ->setConstructorArgs(['test_field'])
```
Comment (murnieza) [inline tests/Query/NestedQueryTest.php]: Does not look like intentional changes.

Comment (trandangtri) [inline tests/Query/NestedQueryTest.php]: Automatic reformatting code in my IDE. I turned off.

**Comment by trandangtri** [comment]
Hi guys,

My fix failed in an old test case. But in my opinion, that test case is not good enough. Look at:

https://github.com/ongr-io/ElasticsearchDSL/blob/master/tests/Query/BoolQueryTest.php#L75

The expected result is

```php
$expected = [
    'should'   => [
        [
            'term' => [
                'key1' => 'value1',
            ],
        ],
    ],
    'must'     => [
        [
            'term' => [
                'key2' => 'value2',
            ],
        ],
    ],
    'must_not' => [
        [
            'term' => [
                'key3' => 'value3',
            ],
        ],
    ],
];
```
But it should be like this

```php
$expected = [
    'should'   => [
        'term' => [
            'key1' => 'value1',
        ],
    ],
    'must'     => [
        'term' => [
            'key2' => 'value2',
        ],
    ],
    'must_not' => [
        'term' => [
            'key3' => 'value3',
        ],
    ],
];
```

Please take on a look official document of [ES Bool Query](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-bool-query.html) and their example.

So I do suggest that we should update the old test case.

**Comment by murnieza** [comment]
Actually both cases are valid, because you can have either array of term queries or single query.

I do not see any problems with changing old tests as building logic is changing.

**Comment by trandangtri** [comment]
@murnieza, yes, both of them will work well with ES, but the output of `toArray()` method has one result. That's why I want to update the expected of the old test case, because it's the best.

**Comment by trandangtri** [comment]
I reverted my work. Now we will always keep the array structure of term query in the output of `toArray`. With that way, we don't need to change the old test case.

File: src/Query/BoolQuery.php [hunk 2]
```diff
@@ -111,7 +111,13 @@ public function toArray()
             $mustContainer = $this->container[self::MUST];
             $query = array_shift($mustContainer);
 
-            return [$query->getType() => $query->toArray()];
+            if ($query->getType() == 'match') {
```
Comment (saimaz) [inline src/Query/BoolQuery.php]: This won't fix the bug. The match query is just like an example of any query. This bug occurs on all other queries also.

**Comment by saimaz** [comment]
For the test cases please write a data provider, don't need to dublicate it. btw, when you will use data provider please include in addition also different queries/filters, not only match. Match query was like an example of query, the bug occurs on any query or filter.

**Comment by saimaz** [comment]
fixed in #51

**Pull request closed without merging**